File: core/src/com/sheepsgohome/GameTools.java
package com.sheepsgohome;

import com.badlogic.gdx.math.Vector2;
import com.badlogic.gdx.physics.box2d.Body;

import java.util.Random;

/**
 * Created by LittleLight on 10.1.2015.
 */
public class GameTools {
    public static Random random = new Random();


    private static float getRandomFloat() {
        return (random.nextFloat() * 2f) - 1f;
    }

    public static void setRandomMovement(Body b, float speed) {
        b.setLinearVelocity(getRandomFloat() * speed,getRandomFloat() * speed);
    }

    public static float calculateAngle(Vector2 v) {
        return calculateAngle(v.x, v.y);
    }


    public static float calculateAngle(float targetx, float targety) {
        float angle = (float) Math.toDegrees(Math.atan2(targety, targetx));

        if (angle < 0) {
            angle += 360;
        }

        return angle;
    }


}

File: core/src/com/sheepsgohome/GameTools.kt
package com.sheepsgohome

import com.badlogic.gdx.math.Vector2
import com.badlogic.gdx.physics.box2d.Body
import java.util.*

object GameTools {
    var random = Random()


    private val randomFloat: Float
        get() = random.nextFloat() * 2f - 1f

    fun setRandomMovement(b: Body, speed: Float) {
        b.setLinearVelocity(randomFloat * speed, randomFloat * speed)
    }

    fun calculateAngle(v: Vector2) = calculateAngle(v.x, v.y)


    fun calculateAngle(targetx: Float, targety: Float): Float {
        var angle = Math.toDegrees(Math.atan2(targety.toDouble(), targetx.toDouble())).toFloat()

        if (angle < 0) {
            angle += 360f
        }

        return angle
    }


}
